k2/csrc/cuda/array.h

```diff
@@ -322,16 +327,18 @@ class Array2 {
   /* Create new array2 with given dimensions.  dim0 and dim1 must be >0.
      Data will be uninitialized. */
   Array2(ContextPtr c, int32_t dim0, int32_t dim1)
-      : dim0_(dim0), dim1_(dim1), elem_stride0_(dim1), byte_offset_(0) {
+      : dim0_(dim0), elem_stride0_(dim1), dim1_(dim1), byte_offset_(0) {
+    K2_DCHECK_GT(dim0, 0);
```

Comment:
BTW, for those index check, which version should we use? `DCHECK` or `CHECK`? I'd like to use `CHECK` if it's not in a loop, and `DCHECK` if it's in a loop. What do you think? I'm just thinking it's better if we can follow a same rule.

Reply:
I am fine with either case. I just replaced all `assert`  that I could find with debug check.

Reply:
yes, I know, just to make sure we can follow the same convention in the future, @danpovey any suggestion?

Reply:
@qindazhu I'd like to use K2_CHECK if it's not in an inner loop or expensive.  If somewhat expensive then K2_CHECK, else K2_PARANOID_ASSERT (or we could implement K2_PARANOID_CHECK).

Also i want to remove glog and just use the K2_ macros, this will keep things simple and reduce chance of future problems.  We need to remove the glog include from cuda/context.h.

Reply:
Yeah, by saying `CHECK` I mean `K2_CHECK` (sorry for the words as I suppose we will remove glog).

> If somewhat expensive then K2_CHECK, else K2_PARANOID_ASSERT

I don't get what you means here, when expensive, still use `K2_CHECK`? (I just thought it's `K2_DCHECK`)